vmb/matrimony/models/profiles.py

```diff
@@ -976,7 +976,7 @@ class Meta:
     def save(self, *args, **kwargs):
         if self.primary:
             try:
-                temp = Photo.objects.get(primary=True)
+                temp = Photo.objects.get(primary=True, profile=self.profile)
```

Comment:
Remove these changes from this PR. Let's fix it in another PR, for which there is already an issue.